### ISSUE TITLE
ApiKeyRequired not allowed in Auth section of AWS::Serverless::Api resource

### ISSUE THREAD

**Issue by RadicalDev** [aws-cloudformation/cfn-python-lint#1159]
*cfn-lint version: 0.24.4*

*CFN lint complains about ApiKeyRequired being specified in the Auth section of AWS::Serverless::Api resources.*

According to [this documentation](https://github.com/awslabs/serverless-application-model/blob/master/versions/2016-10-31.md#api-auth-object) as well as the [Example: Defining API Keys](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/serverless-controlling-access-to-apis.html) section of the tutorials, ApiKeyRequired is a supported item in the Auth section. 

cfn-lint, however, is complaining about its presence with the error message:
*E0001 Error transforming template: Resource with id [Api] is invalid. Invalid value for 'Auth' property*

Example template that cfn-lint 0.24.4 complains about:
```yaml
AWSTemplateFormatVersion: 2010-09-09

Transform: 'AWS::Serverless-2016-10-31'

Resources:
  Api:
    Type: AWS::Serverless::Api
    Properties:
      StageName: api
      EndpointConfiguration: EDGE
      TracingEnabled: true
      Auth:
        ApiKeyRequired: true
```

Incidentally, it doesn't seem possible to ignore this error, either.

**Comment by kddejong** [comment]
@RadicalDev what version are you using?  I'm not able to replicate this issue with the most recent version.

**Comment by kddejong** [comment]
What version of `aws-sam-translator` are you using?

I haven't kept up with up ticking the version because the SAM CLI has been using an older version.  I'm on `1.15.0` but looks like `1.15.1` is out there now too.  
https://github.com/awslabs/aws-sam-cli/blob/v0.23.0/requirements/base.txt#L9

I've created pull request #1166 to require the most recent version but want to validate that this is the issue.

**Comment by kddejong** [comment]
@RadicalDev can you let me know if the newest version of cfn-lint fixes this issue?

**Comment by kddejong** [comment]
This should be good with a newer version of cfn-lint and sam-translator.  We can re-open if its still broken.
